Individual.cpp

```diff
@@ -1441,70 +1440,68 @@ movedata Individual::smsMove(Landscape* pLand, Species* pSpecies,
 					if (pCell == 0) nbr.cell[x2][y2] = 0.0; // no-data cell
 				}
 			}
-
-			sum_nbrs += nbr.cell[x2][y2];
-		}
-	}
-
-	// scale effective costs as probabilities summing to 1
-	if (sum_nbrs > 0.0) { // should always be the case, but safest to check...
-		for (y2 = 2; y2 > -1; y2--) {
-			for (x2 = 0; x2 < 3; x2++) {
-				nbr.cell[x2][y2] = nbr.cell[x2][y2] / (float)sum_nbrs;
-			}
 		}
 	}
 
-	// set up cell selection probabilities
+	// sum up cell weights
 	double cumulative[9];
 	int j = 0;
-	cumulative[0] = nbr.cell[0][0];
 	for (y2 = 0; y2 < 3; y2++) {
 		for (x2 = 0; x2 < 3; x2++) {
-			if (j != 0) cumulative[j] = cumulative[j - 1] + nbr.cell[x2][y2];
+			assert(nbr.cell[x2][y2] >= 0);
+			sum_nbrs += nbr.cell[x2][y2];
+			cumulative[j] = sum_nbrs;
 			j++;
 		}
 	}
 
+	// scale cumulative weights to convert them to cumulative probabilities
+	if (sum_nbrs > 0.0) { // should always be the case, but safest to check...
+		for (j = 0; j < 9; j++) {
+			cumulative[j] /= sum_nbrs;
+		}
+		assert(cumulative[8] == 1.);
+	}
+	else {
+		// unable to make a move
+		// flag individual to die
+		move.dist = -123.0;
+		pCurrCell = 0;
+		return move;
```

Comment:
note to myself: I need to check whether an individual should indeed die if all neighbouring cells have a weight of zero

Reply:
Answer: No, in my understanding it may occur that all cells have a weight of zero but be suitable, in which case the individual will pick a cell randomly among all the equally unappealing but livable cells, and live.